zyte_common_items/base.py

```diff
@@ -126,9 +127,7 @@ def _apply_field_types_to_sub_fields(cls, item: Dict, trail: _Trail = None):
         """
         from_dict, from_list = {}, {}
 
-        annotations = ChainMap(
-            *(c.__annotations__ for c in cls.__mro__ if "__annotations__" in c.__dict__)
-        )
+        annotations = ChainMap(*(get_type_hints(c) for c in cls.__mro__))
```

Comment:
Got hit by [PEP-649](https://peps.python.org/pep-0649/) due to the forward reference in `AEProduct.hasVariants`, glad https://github.com/scrapinghub/scrapy-poet/pull/188 was fresh in my memory :slightly_smiling_face: